radicle-cob/src/sign/message.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2019-2020 The Radicle Foundation <hello@radicle.foundation>
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use git_trailers::Trailer;
+
+use std::fmt::{self, Display};
+
+pub struct CommitMessage<'a> {
+    body: &'a str,
+    trailers: Vec<Trailer<'a>>,
+}
+
+impl<'a> CommitMessage<'a> {
+    pub fn new<I, It>(body: &'a str, trailers: It) -> Self
+    where
+        I: Iterator<Item = Trailer<'a>>,
+        It: IntoIterator<Item = Trailer<'a>, IntoIter = I>,
+    {
+        Self {
+            body,
+            trailers: trailers.into_iter().collect(),
+        }
+    }
+}
+
+impl Display for CommitMessage<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}\n\n", self.body)?;
+
+        for (i, trailer) in self.trailers.iter().enumerate() {
+            write!(f, "{}", trailer.display(": "))?;
+            if i < self.trailers.len() {
+                f.write_str("\n")?;
+            }
+        }
+
+        Ok(())
+    }
+}
```

Comment:
Can we use headers for this?

Reply:
Hmmm true, we probably could. I'd feel nervous about using headers since they're not really established in any documentation and seem to be part of internal `git` whereas this is exactly what trailers are for.

I'm happy to shift to headers, but just wanted to convey the above at least :)

Reply:
Yeah good point. No real issue with trailers, the only thing is that it means we have header parsing code as well as trailer parsing code, but it's not such a big deal. Let's just try to keep it in one place somewhere.

Reply:
I was thinking I could add trailers to `git-commit` so they can be parsed there

Reply:
Yes great idea. There's some code here that may be useful as well: https://github.com/radicle-dev/heartwood/blob/master/radicle/src/storage/git.rs#L656 -- I think I'll probably delete it once we move the sigs to the headers.